src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2541,6 +2541,32 @@ public static void NJulianRuleTest(string posixRule, int dayNumber, int monthNum
             }
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public static void TimeZoneInfo_LocalZoneWithInvariantMode()
+        {
+            bool runningInInvariantMode = false;
+            try { CultureInfo.GetCultureInfo("en-US"); } catch { runningInInvariantMode = true; }
+
+            string hostTZId = TimeZoneInfo.Local.Id;
+
+            ProcessStartInfo psi = new ProcessStartInfo() {  UseShellExecute = false };
+            psi.Environment.Add("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT", runningInInvariantMode ? "0" : "1");
```

Comment:
Should we instead have a test that ensures we are always running on Invariant mode and one that doesn't rather than depending on the host configuration?

Reply:
This test written in a way we always ensure globalization invariant mode is enabled either on the host or on the spawned process. I manually tested that on different environment to ensure it is behaving well. Waiting the CI results too to see if there is any case this test might fail :-)

Reply:
Sounds good 😄

Reply:
We have used `PlatformDetection.IsInvariantGlobalization` in other places.